packages/ic-http-gateway/src/protocol/handler.rs

```diff
@@ -18,9 +18,16 @@ use ic_utils::{
 };
 
 fn convert_request(request: CanisterRequest) -> HttpGatewayResult<HttpRequest> {
+    let uri = request.uri();
+    let mut url = uri.path().to_string();
+    if let Some(query) = uri.query() {
+        url.push('?');
```

Comment:
How does the URI crate handles the query? does it return it without the `?`? if yes, would be nice to add a test for that. 

The behaviour on this things in JS for example is to return with the `?`.

Reply:
Good question. It's returning it without the `?`, which is a bit strange to be honest. I've added a test now to verify the behaviour.